packages/headless-react/src/ssr-commerce/providers.tsx

```diff
@@ -46,19 +48,31 @@ export function buildProviderWithDefinition(looseDefinition: LooseDefinition) {
     useEffect(() => {
       const {searchActions, controllers} = staticState;
 
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      const hydrateControllers: Record<string, any> = {};
+      const hydrateControllers: Record<string, object> = {};
 
-      if ('cart' in controllers) {
-        hydrateControllers.cart = {
-          // eslint-disable-next-line @typescript-eslint/no-explicit-any
-          initialState: {items: (controllers as any).cart.state.items},
-        };
-      }
+      for (const [key, controller] of Object.entries(controllers)) {
+        const typedController = controller as ControllerWithKind;
+
+        switch (typedController._kind) {
+          case Kind.Cart:
+            hydrateControllers[key] = {
+              // eslint-disable-next-line @typescript-eslint/no-explicit-any
+              initialState: {items: (controllers as any)[key].state.items},
+            };
+            break;
+
+          case Kind.Context:
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            hydrateControllers[key] = (controllers as any)[key].state;
+
+            break;
+
+          case Kind.ParameterManager:
+            // TODO
+            break;
```

Comment:
I am planning to wait until https://github.com/coveo/ui-kit/pull/4770 gets merged and rewrite this part

Reply:
It's now in the merge queue!